docs/source/x_ext.rst

```diff
@@ -531,13 +503,83 @@ The |processor| shall cause an illegal instruction fault when attempting to exec
+  | **Signal**             | **Type**                 | **Description**                                                                                                 |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``id``                 | :ref:`id_t <id>`         | Identification of the offloaded instruction.                                                                    |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``rs[X_NUM_RS-1:0]``   | logic [X_RFR_WIDTH-1:0]  | Register file source operands for the offloaded instruction.                                                    |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``rs_valid``           | :ref:`registerflags_t    | Validity of the register file source operand(s). If register pairs are supported, the validity is signaled for  |
+  |                        | <registerflags>`         | each register within the pair individually.                                                                     |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``ecs``                | logic [5:0]              | Extension Context Status ({``mstatus.xs``,``mstatus.fs``,``mstatus.vs``}).                                      |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``ecs_valid``          | logic                    | Validity of the Extension Context Status.                                                                       |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+
+There are two main scenarios, in how the register interface will be used:
+
+1. A register transaction can be started in the same clock cycle as the issue transaction (``issue_req_valid = register_valid`` and ``issue_req.id = register.id``).
+   In this case, the |processor| will speculatively provide all possible source registers via ``register.rs`` when they become available (signalled via the respective ``valid`` signals). 
+   The |coprocessor| will delay accepting the instruction until all necessary registers are provided, and only then assert ``issue_ready``.
+   The ``rs_valid`` bits are not required to be stable during the transaction.
```

Comment:
Here it is not clear what 'transaction' means because this interface does not have its own ready signal. You could say transaction is from register_valid to issue_ready.

Why not simply add a register_ready signal and add the following restrictions for the two scenarios:

1) issue_valid == register_valid && issue_ready == register_ready
2) issue_ready == 1

Reply:
agree

Reply:
fixed